lxd/auth/oidc/oidc.go

```diff
@@ -791,3 +792,20 @@ func NewVerifier(issuer string, clientID string, clientSecret string, scopes []s
 
 	return verifier, nil
 }
+
+// ValidateIssuer performs OIDC discovery to ensure the configuration is valid.
+// It use Discover function to check if given issuer is valid.
+func (o *Verifier) ValidateIssuer(ctx context.Context) error {
+	httpClient, err := o.httpClientFunc()
+	if err != nil {
+		return fmt.Errorf("Failed to get a HTTP client: %w", err)
+	}
+
+	// As this function concentrates on validating the configuration, we do not store the discovered configuration.
+	_, err = client.Discover(ctx, o.issuer, httpClient)
+	if err != nil {
+		return fmt.Errorf("Failed to discover OIDC configuration: %w", err)
+	}
+
+	return nil
+}
```

Comment:
The zitadel library only needs to perform discovery once. After that, the JWKs are stored in a "remote key set" which will automatically pull new ones from the issuer as needed.

Because of this, I think it's wasteful to perform discovery more than once. I think we can remove this function and simply have `NewVerifier` call `ensureConfig` after the fields are instantiated - returning any error.